generic/tests/ping.py

```diff
@@ -1,133 +1,182 @@
-            cmd.append(ext_host)  # external host
+    if ping_ext_host:
+        error_context.context("Ping test from guest side,"
+                              " dest: '%s'" % ext_host, logging.info)
 
+        for size in packet_sizes:
             if params.get("os_type") == "windows":
-                cmd.append("-n 10")
-                cmd.append("-l %s")
-                # Windows doesn't support ping with packet
-                # larger than '65500'
-                pkt_sizes = [p for p in packet_sizes if p < 65500]
-                # Add a packet size just equal '65500' for windows
-                pkt_sizes.append(65500)
+                logging.info("Ping with packet size: %s" % size)
+                status, output = _ping_with_params(packet_size=size,
+                                                   session=session)
+                _check_result(status, output)
             else:
-                cmd.append("-c 10")  # ping 10 times
-                cmd.append("-s %s")  # packet size
-            cmd = " ".join(cmd)
-            for size in pkt_sizes:
-                error.context("Ping with packet size %s" % size,
+                for interval in interval_times:
+                    logging.info("Ping with packet size: %s and interval: %s" %
+                                 (size, interval))
+                    status, output = _ping_with_params(packet_size=size,
+                                                       session=session,
+                                                       interval=interval)
+                    _check_result(status, output)
+
+        # ping with exceeded packages
+        error_context.context("Ping test with exceeded package",
                               logging.info)
-                status, output = session.cmd_status_output(cmd % size,
-                                                           timeout=60)
-                _get_loss_ratio(output)
-
-                if status != 0:
-                    raise error.TestFail(("Ping external host failed,"
-                                          " status: %s, output: %s" %
-                                          (status, output)))
+        status, output = _ping_with_params(packet_size=65508, session=session)
+        if output.find(result_str) == -1:
+            test.fail(("Ping test with exceeded package failed,"
+                       " expected string: '%s' not found" %
+                       result_str))
+
+        # ping with flood
+        error_context.context("Flood ping test from guest", logging.info)
+
+        if params.get("os_type") == "windows":
+            """
+            Under Windows, we have to use a dedicated session because the
+            aexpect does not have the other method to interrupt the process
+            in the guest rather than close the session. And we should use
+            hard code to avoid process terminate exception in utils_net.ping()
+            funciton.
+            """
```

Comment:
Maybe we can kill it through another session?

Reply:
Please comment it instead of using docstring because it is just a comment.

Reply:
I will try this and modify it according to your advice .
Many thanks